baseplate/core.py

```diff
@@ -169,14 +182,6 @@ def _secret(self):
 
         return self._secret_store.get_simple("jwt/authentication/secret")
 
-    def attach_context(self, context):
-        """Attach this authentication wrapper to the provided context
-
-        :param context: request context to attach this authentication to
-
-        """
-        context.authentication = self
-
```

Comment:
How about I put the helper method back in with a `warn_deprecated`?  Since the header is forwarded from the `context.request_context` attribute, I would prefer to discourage use of the `AuthenticationContext.attach_context` method.  If no one is currently using `context.authentication` I would rather just remove it to avoid any confusion or mis-use.

Reply:
I re-added this code, I think we should consider removing it at some point though.

Reply:
warn_deprecated sounds fine. We're technically at a 0 major version still so breaking changes are "OK" but I really like to avoid them when at all possible because it's more friendly to open source and makes it easier to upgrade production services.